File: taint-fixtures-v2/python/cwe89/tn_parameterized_query.py
# Fixture: code_patch · CWE-89 SQL Injection · Python
# VERDICT: TRUE_NEGATIVE
# PATTERN: sql_parameterized_query
# SOURCE: http_request (request.args)
# SINK: cursor.execute (parameterized)
# TAINT_HOPS: 1
# NOTES: Parameterized query — value is bound, not interpolated
import sqlite3
from flask import request


def get_user():
    username = request.args.get("username")
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    # SAFE: parameterized query — user input is a bound parameter
    cursor.execute("SELECT * FROM users WHERE username = ?", (username,))
    return cursor.fetchone()

File: taint-fixtures-v2/go/cwe89/tn_parameterized_query.go
// Fixture: code_patch · CWE-89 SQL Injection · Go
// VERDICT: TRUE_NEGATIVE
// PATTERN: sql_parameterized_query
// SOURCE: http_request (r.URL.Query)
// SINK: db.QueryRow (parameterized)
// TAINT_HOPS: 1
// NOTES: Parameterized query — placeholder prevents injection
package db

import (
	"database/sql"
	"net/http"
)

func GetUserSafe(database *sql.DB, r *http.Request) *sql.Row {
	username := r.URL.Query().Get("username")
	// SAFE: parameterized query with placeholder
	return database.QueryRow("SELECT id, username, email FROM users WHERE username = ?", username)
}
